homeassistant/components/tile/device_tracker.py

```diff
@@ -1,140 +1,102 @@
-"""Support for Tile® Bluetooth trackers."""
-from datetime import timedelta
-import logging
+"""Support for Tile device trackers."""
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
+from homeassistant.components.device_tracker.const import SOURCE_TYPE_GPS
+from homeassistant.core import callback
 
-from pytile import async_login
-from pytile.errors import SessionExpiredError, TileError
-import voluptuous as vol
-
-from homeassistant.components.device_tracker import PLATFORM_SCHEMA
-from homeassistant.const import CONF_MONITORED_VARIABLES, CONF_PASSWORD, CONF_USERNAME
-from homeassistant.helpers import aiohttp_client, config_validation as cv
-from homeassistant.helpers.event import async_track_time_interval
-from homeassistant.util import slugify
-from homeassistant.util.json import load_json, save_json
-
-_LOGGER = logging.getLogger(__name__)
-CLIENT_UUID_CONFIG_FILE = ".tile.conf"
-DEVICE_TYPES = ["PHONE", "TILE"]
+from . import DATA_COORDINATOR, DOMAIN, TileEntity
 
 ATTR_ALTITUDE = "altitude"
 ATTR_CONNECTION_STATE = "connection_state"
 ATTR_IS_DEAD = "is_dead"
 ATTR_IS_LOST = "is_lost"
 ATTR_RING_STATE = "ring_state"
 ATTR_VOIP_STATE = "voip_state"
-ATTR_TILE_ID = "tile_identifier"
 ATTR_TILE_NAME = "tile_name"
 
-CONF_SHOW_INACTIVE = "show_inactive"
 
-DEFAULT_ICON = "mdi:view-grid"
-DEFAULT_SCAN_INTERVAL = timedelta(minutes=2)
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Tile device trackers."""
+    coordinator = hass.data[DOMAIN][DATA_COORDINATOR][config_entry.entry_id]
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_SHOW_INACTIVE, default=False): cv.boolean,
-        vol.Optional(CONF_MONITORED_VARIABLES, default=DEVICE_TYPES): vol.All(
-            cv.ensure_list, [vol.In(DEVICE_TYPES)]
-        ),
-    }
-)
+    async_add_entities(
+        [
+            TileDeviceTracker(coordinator, tile_uuid, tile)
+            for tile_uuid, tile in coordinator.data.items()
+        ],
+        True,
+    )
 
 
-async def async_setup_scanner(hass, config, async_see, discovery_info=None):
```

Comment:
Is this a breaking change?

Can we keep `async_setup_scanner` and import the config entry from there instead of adding new schema?

I'm not 100% sure it will work with a device tracker, but I was able to do something similar in `harmony` (remote.py) and `nut`(sensor.py) to avoid the need to avoid a breaking change.

Reply:
I'm not terribly familiar with the legacy device tracker framework – is maintaining this method necessary for existing users of this platform?

Reply:
I don't think so.  I'm suggesting import the config entry here so they don't have to re-setup.

Reply:
Bear with me – I'm wondering if this is necessary.

With this PR, `async_setup` inside `tile/__init__.py` imports the Tile config from`configuration.yaml`:

https://github.com/home-assistant/core/blob/368e46052edb5830a53bbb4e4b85008c5a3909ae/homeassistant/components/tile/__init__.py#L53-L68

This should do all the setup that's necessary, even for existing Tile consumers.

Am I missing something?

Reply:
So if I setup tile https://www.home-assistant.io/integrations/tile/
```
device_tracker:
 - platform: tile
   username: nick@koston.org
   password: SNIP
```

Its not going to be able to import the above since the schema is now 

```
tile:
   username: nick@koston.org
   password: SNIP
```

Reply:
Took a deeper look at this and unfortunately, it can't be done cleanly for device trackers (since their legacy support is so complicated). I'm thinking a breaking change is the only way to go here. I'll chew on it a bit more and get back to you.

Reply:
Okay, you were right – check out 3411f68.

That said, I still see a breaking change that I didn't previously: this PR updates the entity IDs to human-friendly strings; since this is the first time Tile makes use of the entity registry, they wouldn't have been in there before.

Reply:
Does that mean we can drop the new CONFIG_SCHEMA as well?

Reply:
Man, good thing you're here, @bdraco. 😆  Yes; will remove.